src/lib/comms/sol-coap.c

```diff
@@ -1072,7 +1072,7 @@ sol_coap_packet_new_request(sol_coap_method_t method, sol_coap_msgtype_t type)
 }
 
 SOL_API int
-sol_coap_add_option(struct sol_coap_packet *pkt, uint16_t code, const void *value, uint16_t len)
+sol_coap_add_option(struct sol_coap_packet *pkt, uint16_t code, const void *value, size_t len)
```

Comment:
I also prefer to document on .h's for those with 16bit limits then to change the type